app/models/manageiq/providers/google/cloud_manager/refresh_parser.rb

```diff
@@ -0,0 +1,189 @@
+require 'fog/google'
+
+module ManageIQ::Providers
+  module Google
+    class CloudManager::RefreshParser < ManageIQ::Providers::CloudManager::RefreshParser
+      include Vmdb::Logging
+
+      def self.ems_inv_to_hashes(ems, options = nil)
+        new(ems, options).ems_inv_to_hashes
+      end
+
+      def initialize(ems, options = nil)
+        @ems               = ems
+        @connection        = ems.connect
+        @options           = options || {}
+        @data              = {}
+        @data_index        = {}
+      end
+
+      def ems_inv_to_hashes
+        log_header = "Collecting data for EMS : [#{@ems.name}] id: [#{@ems.id}]"
+
+        _log.info("#{log_header}...")
+        get_zones
+        get_flavors
+        get_cloud_networks
+        get_images
+        get_instances
+        _log.info("#{log_header}...Complete")
+
+        @data
+      end
+
+      private
+
+      def get_zones
+        zones = @connection.zones.all
+        process_collection(zones, :availability_zones) { |zone| parse_zone(zone) }
+      end
+
+      def get_flavors
+        flavors = @connection.flavors.all
+        process_collection(flavors, :flavors) { |flavor| parse_flavor(flavor) }
+      end
+
+      def get_cloud_networks
+        networks = @connection.networks.all
+        process_collection(networks, :cloud_networks) { |network| parse_cloud_network(network) }
+      end
+
+      def get_images
+        images = @connection.images.all
+        process_collection(images, :vms) { |image| parse_image(image) }
+      end
+
+      def get_instances
+        instances = @connection.servers.all
+        process_collection(instances, :vms) { |instance| parse_instance(instance) }
+      end
+
+      def process_collection(collection, key)
+        @data[key] ||= []
+
+        collection.each do |item|
+          uid, new_result = yield(item)
+          next if uid.nil?
+
+          @data[key] |= [new_result]
+          @data_index.store_path(key, uid, new_result)
+        end
+      end
+
+      def parse_zone(zone)
+        name = uid = zone.name
+        type = ManageIQ::Providers::Google::CloudManager::AvailabilityZone.name
+
+        new_result = {
+          :type    => type,
+          :ems_ref => uid,
+          :name    => name,
+        }
+
+        return uid, new_result
+      end
+
+      def parse_flavor(flavor)
+        uid = flavor.name
+
+        type = ManageIQ::Providers::Google::CloudManager::Flavor.name
+        new_result = {
+          :type        => type,
+          :ems_ref     => flavor.name,
+          :name        => flavor.name,
+          :description => flavor.description,
+          :enabled     => !flavor.deprecated,
+          :cpus        => flavor.guest_cpus,
+          :cpu_cores   => flavor.guest_cpus,
```

Comment:
Is this right? I thought cpus is the number of cpus, and cpu_cores is the cores per cpu

cc @bdunne Looks like another candidate for the better renaming.